starknet_py/net/websockets/websocket_client.py

```diff
@@ -322,11 +427,13 @@ def _handle_received_message(self, message: Union[str, bytes]):
         # case when the message is a response to `subscribe_{method}`
         if "id" in data and data["id"] in self._pending_responses:
             future = self._pending_responses.pop(data["id"])
-            future.set_result(data)
-
-        # case when the message is a notification
+            if not future.done():
+                future.set_result(data)
```

Comment:
What in case it was done? This is non standard condition I think, maybe we should raise? 

https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.set_result

It should raise if it was done, so maybe it makes sense to remove the `if not future.done()` guard?